File: Excepciones/WarningException.vb
Imports Entidades
Public Class WarningException
    Inherits Exception

    Public Property ListaMensajes As List(Of String)
    Public Property MensajesFamPat As List(Of FamPatEN)
    Public Property MensajesUsuFam As List(Of UsuFamEN)

    Public Sub New(ByVal mensaje As String)
        MyBase.New(mensaje)
    End Sub

    Public Sub New(ByVal Mensajes As List(Of FamPatEN))
        MensajesFamPat = Mensajes
    End Sub

    Public Sub New(ByVal Mensajes As List(Of UsuFamEN))
        MensajesUsuFam = Mensajes
    End Sub
End Class

File: Excepciones/WarningException.cs
using System;
using System.Collections.Generic;
using Entidades;

namespace Excepciones
{
    public class WarningException : Exception
    {
        public List<string> ListaMensajes { get; set; }
        public List<FamPatEN> MensajesFamPat { get; set; }
        public List<UsuFamEN> MensajesUsuFam { get; set; }

        public WarningException(string mensaje) : base(mensaje)
        {
        }

        public WarningException(List<FamPatEN> Mensajes)
        {
            MensajesFamPat = Mensajes;
        }

        public WarningException(List<UsuFamEN> Mensajes)
        {
            MensajesUsuFam = Mensajes;
        }
    }
}